File: new_engine/Engine/res/script/Camera.lua
local lib = require("script.Engine");

function MoveCamera()

  local speed = 4;

  local camera = Engine.Scene.Camera;
  local input = Engine.Input;

  if (input.getKey("UP"):isDown()) then
    camera:getPosition().y = camera:getPosition().y + speed;
  end

  if (input.getKey("DOWN"):isDown()) then
    camera:getPosition().y = camera:getPosition().y - speed;
  end

  if (input.getKey("LEFT"):isDown()) then
    camera:getPosition().x = camera:getPosition().x - speed;
  end

  if (input.getKey("RIGHT"):isDown()) then
    camera:getPosition().x = camera:getPosition().x + speed;
  end

end

File: src/engine/entity/Camera.java
package engine.entity;

import org.joml.Vector2f;

public class Camera {

	private Vector2f position;
	// private int width, height;
	private EntitySprite spriteToWatch = null;

	public Camera(Vector2f position, int screenWidth, int screenHeight) {
		this.setPosition(position);
		// this.width = screenWidth;
		// this.height = screenHeight;
	}

	public void update() {
		if (spriteToWatch != null) {
			// position = new
			// Vector2f(spriteToWatch.getPosition()).add(spriteToWatch.getDimensions().mul(0.5f));
		}
		// float speed = 1f;
		// if (Input.getKeyByName("UP").isDown()) {
		// position.y += speed;
		// }
		// if (Input.getKeyByName("DOWN").isDown()) {
		// position.y -= speed;
		// }
		// if (Input.getKeyByName("LEFT").isDown()) {
		// position.x -= speed;
		// }
		// if (Input.getKeyByName("RIGHT").isDown()) {
		// position.x += speed;
		// }
		// // position.x *= 0.9f;
		// // position.y *= 0.9f;
	}

	public void watchSprite(EntitySprite sprite) {
		spriteToWatch = sprite;
	}

	public Vector2f getPosition() {
		return position;
	}

	public void setPosition(Vector2f position) {
		this.position = position;
	}
}
